lib/elixir/lib/gen_server.ex

```diff
@@ -572,6 +572,9 @@ defmodule GenServer do
   end
 
   def call(server, request, timeout) do
+    if match?({:global, _}, server) and GenServer.whereis(server) == self() do
```

Comment:
@TBK145 it would probably be better something like:

    pid = whereis(server)
    if pid == self() do
      raise ...
    end
    :gen.call(pid, :"$gen_call", request, timeout)

i.e. we don't need to guard against global, we can always check it. We can also remove the clause we have defined earlier.

@fishcakez thoughts?